src/Chapter3.hs

```diff
@@ -480,6 +528,60 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data Castle = Castle String 
+    deriving (Show, Eq)
+
+data Wall = Wall Bool 
+    deriving (Show, Eq)
+
+data Community = Church | Library 
+    deriving (Show, Eq)
+
+data Housing = Housing
+    { oneOccupant    :: Int
+    , twoOccupants   :: Int
+    , threeOccupants :: Int
+    , fourOccupants  :: Int
+    } deriving (Show, Eq)
```

Comment:
It's not clear what does this data type represent :thinking: Is it one house? Or the number of houses of each size?

In this task we expected people to implement a data type for representing a single house in a type-safe way :slightly_smiling_face: And then, you will have a list of such houses in the city type.